packages/board-server/src/router.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import type { IncomingMessage, ServerResponse } from "node:http";
+import type { ViteDevServer } from "vite";
+
+import { serveContent } from "./server/common.js";
+import { serveBoardsAPI } from "./server/boards/index.js";
+import { serveProxyAPI } from "./server/proxy/index.js";
+import { serveInfoAPI } from "./server/info/index.js";
+import { serveHome } from "./server/home/index.js";
+
+export function makeRouter(hostname: string, vite: ViteDevServer | null) {
+  return async function router(
+    req: IncomingMessage,
+    res: ServerResponse
+  ): Promise<void> {
+    const url = new URL(req.url || "", hostname);
+
+    if (await serveHome(req, res)) {
```

Comment:
I kinda prefer having the path handling happen here (like with the switch in https://github.com/breadboard-ai/breadboard/blob/main/packages/connection-server/src/router.ts). That way the specific handler doesn't need to care where it is "mounted", and it's easy to see all the top-level routing logic in one place. Plus it's more efficient than trying all of the handlers.

Reply:
Ah sorry, I guess you're doing this incrementally, and this aspect of it is unchanged, so you can ignore this comment/address later.

Reply:
I agree. I want to move all routing logic to look like the Connection Server router.

I want to go piece-by-piece, though, to make sure I don't break something along the way.